docs/code-culture.md

```diff
@@ -164,3 +164,66 @@ Example of good framework log messages (a little bit simplified) can look like e
   "message": "Task $taskId of Pod $podId was killed"
 }
 ```
+
+## Testing
+
+Tests are executable documentation and should be **written, read and maintained** as such. A test tells it's reader a short story about how some part of the system should behave given a certain input. Hundreeds of excelent books and articles are dedicated to writing proper tests so we'll try to avoid repeating them here.
+
+### On Libraries and Styles
+
+We heavily utilize [ScalaTest](http://www.scalatest.org/) as our primary test library. When it comes to the test style we prefer [WordSpec](http://www.scalatest.org/at_a_glance/WordSpec)
+
+```scala
+  // Describe a scope for a subject, in this case: "A Set"
+  "A Set" can { // All tests within these curly braces are about "A Set"
+
+    // Can describe nested scopes that "narrow" its outer scopes
+    "empty" should { // All tests within these curly braces are about "A Set (when empty)"
+
+      "have size 0" in {    // Here, 'it' refers to "A Set (when empty)". The full name
+        Set.empty.size shouldBe 0 // of this test is: "A Set (when empty) should have size 0"
```

Comment:
maybe we could come up with a different example? Because for this case, I think `shouldBe empty` is a better approach :)

Reply:
I know! But I also failed to come up with a better example and left it like this for now